### ISSUE TITLE
Selecting multiple Domains in the uBlock Menu at once

### ISSUE THREAD

**Issue by hipunk** [gorhill/uBlock#1316]
Feature request:

I'd like to be able to select multiple domains/scripts (what do you call the items listed in the screenshot? I'll just call them items for now) at once when being in the menu.
Like in Finder or Explorer: Click one item, shift click another item and all items from first selected item until last select item gets selected.
To decide if you wish to block, unblock or leave it undecided/filterruled you could then just implement it in a way, that the user clicks one item (e.g. click allow on one item) and all the other selected domains also get put into that state (all of the selected items get put in the state of the single clicked item).

This would greatly reduce the time you need to invest when you trust just one single site, but do not want to disable the blocker completely using the big blue button.
To further explain what I mean, here's a screenshot of what menu I'm talking about:
https://i.imgur.com/44HOLjo.png
In this example I have to click 
4 (2) times to allow everything shown.
Instead, I could also somehow click "github.com", then shiftclick "githubapp.com ", then all 4 items (""github.com", "avatars1.githubusercontent.com", "avatars3.githubusercontent.com", "githubapp.com" would be selected (maybe separation of selected and unselected state through different colors or a colored background?) and then I'd click allow on just one item (e.g. "github.com"), which would allow all 4 items at once with just a single click.
This would save me one click if I wanted to allow all 4 (which is what I do anyway, although the middle ones aren't blocked in the screenshot).

I hope you understand what I'm talking about.

Have a nice day,

hipunk

**Comment by gorhill** [comment]
Sorry but I disagree with significantly increasing code complexity (and work load, destabilization) for rather marginal (and questionable) benefits which may only appeal to an ever smaller minority of users.

**Comment by hipunk** [comment]
First of all thanks for your time reading my posts. It is greatly appreciated, just like the work you've already done with this addon in general. You've done the web community a big favour in my eyes.

Okay, here's a better and much much more brilliant idea: Create an option to **disable blocking on just the current tab**.

Try to visit sites like imageshack or photobucket: You need to allow so much stuff (considering you've got the lists activated that block them per default), to visit the site itself, you've gotta reload the page so often - you're spending more time unblocking than looking at the actual page. Especially when this addon gets combined with other addons like RefControl or NoScript, which sometimes also require a pagerefresh.

To work around that issue you implemented a big blue button, I got that principle and like it.

However, this also disables adblocking on tabs that don't need a deactivated adblocker, which is certainly unreasonable for day to day use.
Imagine having to refresh another tab (after having disabled uBlock) so that you can compare two pages next to each other: In case you forgot to enable adblocking again on the page that never made problems in the first place, you would get bombarded with ads again. Or: Sometimes it's not possible to enable the blocker again, because the site you disabled it for utilises dynamic content loading which uBlock would then break again.

A per tab big blue button in uBlock would
- be easy for the user
- be a solution to my problem
- would appeal more than just a few people (I bet)

As far as I know, no other blocker has done this yet, another reason: You'd be first again.

This would save me a lot of time. Really, it would.


Waiting for your reply,

hipun

**Comment by gorhill** [comment]
> Try to visit sites like imageshack or photobucket: You need to allow so much stuff (considering you've got the lists activated that block them per default), to visit the site itself, you've gotta reload the page so often - you're spending more time unblocking than looking at the actual page

I don't understand that part. Don't these sites works just fine with the default filter lists? If not than the solution is to report any breakage issue to the filter list maintainers.
